src/shared/logger/logTracker.ts

```diff
@@ -0,0 +1,86 @@
+/*!
+ * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { waitUntil } from '../utilities/timeoutUtils'
+
+/**
+ * @param logID  Unique ID assigned to a registered log
+ * @param logMessage  A Promise to be resolved into the message written by the logger 
+ */
+export interface LogTrackerRecord {
+    logID: number,
+    logMessage: Promise<string | undefined>,
+}
+
+/** 
+ * The LogTracker allows for the recording of a log's message after it has passed through a Logger class.
+ */
+export class LogTracker {
```

Comment:
is this really a separate class? Appears simple enough and coupled enough to be part of the Logger module and/or class. Separating adds more cognitive overhead than it saves, maybe, because it implies that it is decoupled?

Reply:
I thought about this more and I think it can be included with `Logger` as long as there is a way to identify which file transport the log should be tracked in. This would be a parameter to `registerLog`, while `logToFile` will return the identifier assigned.

Reply:
> Edit: probably will just be easier to use the file path as the identifier. Having some other way to identify transports would be needed if we wanted to track messages in the consoles.

Maybe Winston has a "tracker id" ? If not, then paths should work, or we can invent a scheme like `type://path`, e.g. `file://path/to/log.txt` or `terminal://<PID>`.